### ISSUE TITLE
`consistent-boolean-name`: add a way to ignore variables created from destructuring?

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3316]
### Description

Thanks for adding such a brilliant rule - I was dreaming of having enforcement like this for a long time.

I tried running it on medium+ sized projects and observed that the majority of the reports come from variables created when destructuring object properties, which in turn we oftentimes do not have any control over. Even if we do, it's either intentional (example: public library APIs shorten names for ergonomics) or something that we don't control (external libraries, APIs, etc).

If you really think about it, this might be very closely related to the enforcement on object properties controlled by the `checkProperties` option. Therefore, my suggestion is to change the rule in either of the following ways:
- Do not check destructuring-created variables unless `checkProperties` is `true`;
- Do not check unless a new option, say, `checkDestructuring`, is `true`.

### Examples

Allow the following by default:

```ts
declare const options: {strict: boolean}

const {strict} = options;
```
